libkbfs/folder_branch_ops.go

```diff
@@ -3960,6 +3950,36 @@ func (fbo *folderBranchOps) notifyOneOpLocked(ctx context.Context,
 	afterUpdateFn func() error) error {
 	fbo.headLock.AssertLocked(lState)
 
+	// We need to get unlinkPath before calling UpdatePointers so that
+	// nodeCache.Unlink can properly update cachedPath.
+	unlinkPath, toUnlink := func() (path, bool) {
+		var (
+			node      Node
+			childName string
+		)
+
+		switch realOp := op.(type) {
+		case *rmOp:
+			node = fbo.nodeCache.Get(realOp.Dir.Unref.Ref())
+			childName = realOp.OldName
+		case *renameOp:
+			node = fbo.nodeCache.Get(realOp.NewDir.Unref.Ref())
+			childName = realOp.NewName
+		}
+		if node == nil {
+			return path{}, false
+		}
+
+		p, err := fbo.pathFromNodeForRead(node)
+		if err != nil {
+			fbo.log.CErrorf(ctx,
+				"Calling pathFromNodeForRead error: %v", err)
```

Comment:
OK sounds good. I guess it wouldn't matter, but just to make sure, should `UpdatePointers` still be called in case of the error? The existing code would have called `UpdatePointers` in case of that error.